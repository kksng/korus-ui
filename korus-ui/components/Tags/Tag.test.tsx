// @ts-nocheck
import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import { Tag, Tags } from './index';

describe('Tag components test:', () => {
  it('should be defined', () => {
    expect(Tag).toBeDefined();
  });
  it('should render', () => {
    const wrapper = mount(
      <Tags>
        <Tag>test</Tag>
      </Tags>,
    );

    expect(wrapper.text()).toEqual('test');

    expect(wrapper.find('Tag span').first().hasClass('tags-item')).toBeTruthy();

    expect(wrapper.find('i.sbicon-times.txt-gray')).toBeTruthy();

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should call onClick callback', () => {
    const handleClick = jest.fn();
    const wrapper = mount(
      <Tags>
        <Tag onClick={handleClick}>test</Tag>
      </Tags>,
    );

    wrapper.find('Tag').prop('onClick')();

    expect(handleClick).toHaveBeenCalled();

    expect(wrapper).toMatchSnapshot();
  });
});
