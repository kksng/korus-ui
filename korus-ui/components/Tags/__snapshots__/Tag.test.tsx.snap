// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag components test: should call onClick callback 1`] = `
<Tags>
  <Wrapper
    className="tags-wrapper"
  >
    <Div
      className="tags-wrapper"
    >
      <div
        className="tags-wrapper"
      >
        <Tag
          key=".0"
          onClick={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            }
          }
          theme={
            Object {
              "closeIcon": "tags-icon",
              "defaultIcon": "icon-default",
              "tag": "tags-item",
              "wrapper": "tags-wrapper",
            }
          }
        >
          <Wrapper
            className="tags-item"
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                ],
              }
            }
          >
            <Span
              className="tags-item"
              onClick={
                [MockFunction] {
                  "calls": Array [
                    Array [],
                  ],
                  "results": Array [
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                  ],
                }
              }
            >
              <span
                className="tags-item"
                onClick={
                  [MockFunction] {
                    "calls": Array [
                      Array [],
                    ],
                    "results": Array [
                      Object {
                        "type": "return",
                        "value": undefined,
                      },
                    ],
                  }
                }
              >
                test
                <Icon
                  className="tags-icon icon-default"
                >
                  <Span
                    className="tags-icon icon-default"
                  >
                    <span
                      className="tags-icon icon-default"
                    />
                  </Span>
                </Icon>
              </span>
            </Span>
          </Wrapper>
        </Tag>
      </div>
    </Div>
  </Wrapper>
</Tags>
`;

exports[`Tag components test: should render 1`] = `
<Tags>
  <Wrapper
    className="tags-wrapper"
  >
    <Div
      className="tags-wrapper"
    >
      <div
        className="tags-wrapper"
      >
        <Tag
          key=".0"
          theme={
            Object {
              "closeIcon": "tags-icon",
              "defaultIcon": "icon-default",
              "tag": "tags-item",
              "wrapper": "tags-wrapper",
            }
          }
        >
          <Wrapper
            className="tags-item"
          >
            <Span
              className="tags-item"
            >
              <span
                className="tags-item"
              >
                test
                <Icon
                  className="tags-icon icon-default"
                >
                  <Span
                    className="tags-icon icon-default"
                  >
                    <span
                      className="tags-icon icon-default"
                    />
                  </Span>
                </Icon>
              </span>
            </Span>
          </Wrapper>
        </Tag>
      </div>
    </Div>
  </Wrapper>
</Tags>
`;
